src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -343,23 +343,70 @@ public enum NodeType
 
         private string StripFormatSpecifier(string exp, out string formatSpecifier)
         {
-            formatSpecifier = null;
+            formatSpecifier = null; // will be used with -var-set-format
             int lastComma = exp.LastIndexOf(',');
-            if (lastComma > 0)
-            {
-                string expFS = exp.Substring(lastComma + 1);
-                string trimmed = expFS.Trim();
-                if (trimmed == "x" || trimmed == "X" || trimmed == "h" || trimmed == "H")
-                {
+            if (lastComma <= 0)
+                return exp;
+
+            string expFS = exp.Substring(lastComma + 1);
+            string trimmed = expFS.Trim();
+            switch (trimmed)
+            {
+                case "x":
+                case "X":
+                case "h":
```

Comment:
would it make sense to call trimmed.toLower() instead? I assume `o` and `O` should be the same specifier. this would get rid of some duplicates?

Reply:
I actually had `string trimmed = expFS.ToLowerInvariant().Trim();` before but removed it because it might cause problems with later usages of `trimmed` and the spec only defines uppercase chars for hex.

Reply:
`switch (trimmed.ToLowerInvariant())` is an option but I don't know if VS accepts uppercase for other specifiers.
On the other hand they seem to allow weird constructs like

https://github.com/inviwo/inviwo/blob/2c7284bdda7e001578c9b39fb508ca19113a7de3/tools/natvis/qt5.natvis#L234